src/XMakeCommandLine/UnitTests/CommandLineSwitches_Tests.cs

```diff
@@ -1153,6 +1156,129 @@ public void ExtractAnyLoggerParameterPickLast()
         }
 
         /// <summary>
+        /// Verifies that when the /warnaserror switch is not specified, the set of warnings is null.
+        /// </summary>
+        [Fact]
+        public void ProcessWarnAsErrorSwitchNotSpecified()
+        {
+            CommandLineSwitches commandLineSwitches = new CommandLineSwitches();
+
+            MSBuildApp.GatherCommandLineSwitches(new ArrayList(new[] { "" }), commandLineSwitches);
+
+            Assert.Null(MSBuildApp.ProcessWarnAsErrorSwitch(commandLineSwitches));
+        }
+
+        /// <summary>
+        /// Verifies that the /warnaserror switch is parsed properly when codes are specified.
+        /// </summary>
+        [Fact]
+        public void ProcessWarnAsErrorSwitchWithCodes()
+        {
+            ISet<string> expectedWarningsAsErors = new HashSet<string>(StringComparer.OrdinalIgnoreCase) { "a", "B", "c", "D", "e" };
+
+            CommandLineSwitches commandLineSwitches = new CommandLineSwitches();
+
+            MSBuildApp.GatherCommandLineSwitches(new ArrayList(new[]
+            {
+                "\"/warnaserror: a,B , c \"", // Leading, trailing, leading and trailing whitespace
+                "/warnaserror:A,b,C",         // Repeats of different case
+                "\"/warnaserror:,    ,,\"",   // Empty items
+                "/err:D,d,E,e",               // A different source with new items and uses the short form
+                "/warnaserror:a",             // A different source with a single duplicate
+                "/warnaserror:a,b",           // A different source with  multiple duplicates
+                "/warnaserror",               // Nothing specified
+            }), commandLineSwitches);
+
+            ISet<string> actualWarningsAsErrors = MSBuildApp.ProcessWarnAsErrorSwitch(commandLineSwitches);
+
+            Assert.NotNull(actualWarningsAsErrors);
+
+            Assert.Equal(expectedWarningsAsErors, actualWarningsAsErrors, StringComparer.OrdinalIgnoreCase);
```

Comment:
For posterity: we discussed, and decided that the additive model makes sense, but a new no-code-specified setting should override previous specific ones. If you need to clear the setting you can do `/warnaserror /warnaserror:X` to get back to a "just X" state.